scripts/build/TestPlatform.Settings.targets

```diff
@@ -6,6 +6,7 @@
     at the start of new iteration to the goal number. This is also used to version the local packages. This version needs to be statically
     readable when we read the file as xml, don't move it to a .props file, unless you change the build server process -->
     <TPVersionPrefix>17.2.0</TPVersionPrefix>
+    <LangVersion>latest</LangVersion>
```

Comment:
We could either stick with a specific version of C#, use `latest` or even use `preview`.

Reply:
I vote for `preview`.

Reply:
are we sure that `preview` won't emit possible bugged IL that we ship?

Reply:
https://github.com/dotnet/runtime/blob/main/Directory.Build.props#L284-L287 runtime repo does some distinction but for cs is preview...but I don't know if there's some override inside csproj, we could take a fast look to the most used lib(HttpClient ie).

Reply:
> are we sure that `preview` won't emit possible bugged IL that we ship?

I can't say 100% sure but the same config you are mentioning is set for `roslyn` and `roslyn-analyzers` so it feels relatively safe to me:
https://github.com/dotnet/roslyn-analyzers/blob/main/Directory.Build.props#L21-L22
https://github.com/dotnet/roslyn/blob/main/eng/targets/Settings.props#L198
https://github.com/dotnet/roslyn/blob/main/eng/targets/Settings.props#L176

Reply:
Ok let's go with it so